Lib/fontParts/base/contour.py

```diff
@@ -456,6 +456,111 @@ def _get_area(self):
+
+        ``otherContour`` must be a :class:`BaseContour`.
+        ``sameStartPoint`` must be a :ref:`type-bool`.
+        ``samePosition`` must be a :ref:`type-bool`.
+        ``samePointType`` must be a :ref:`type-bool`.
+        """
+        otherContour = normalizers.normalizeContour(otherContour)
+        sameStartPoint = normalizers.normalizeBoolean(sameStartPoint)
+        samePosition = normalizers.normalizeBoolean(samePosition)
+        samePointType = normalizers.normalizeBoolean(samePointType)
+        return self._compare(otherContour, sameStartPoint, samePosition, samePointType)
+
+    @staticmethod
+    def _get_pt_distances(contourPoints, samePointType):
+        """
+        Helper method to calcuate distance between points
+        """
+        digest = []
+        for i, point in enumerate(contourPoints):
+            if not samePointType:
+                if i is 0:
+                    digest.append((contourPoints[len(contourPoints)-1][0]-point[0],
+                                  contourPoints[len(contourPoints)-1][1]-point[1]))
+                else:
+                    digest.append((contourPoints[i-1][0]-point[0],
+                                   contourPoints[i-1][1]-point[1]))
+            if samePointType:
+                if i is 0:
+                    digest.append((contourPoints[len(contourPoints)-1][0][0]-point[0][0],
+                                  contourPoints[len(contourPoints)-1][0][1]-point[0][1],
+                                  contourPoints[len(contourPoints)-1][1], point[1]))
+                else:
+                    digest.append((contourPoints[i-1][0][0]-point[0][0],
+                                   contourPoints[i-1][0][1]-point[0][1],
+                                   contourPoints[i-1][1], point[1]))
+        return digest
+
+    @staticmethod
+    def _shift_in_place(digest, n=1):
+        """
+        Helper method to shift digest distances around
+        """
+        n = n % len(digest)
+        head = digest[:n]
+        digest[:n] = []
+        digest.extend(head)
+        return digest
+
+    def _compare(self, otherContour, sameStartPoint, samePosition, samePointType):
+        """
+        Subclasses may override this method.
+        """
+        # First a sanity check for number of points in contours
+        if len(self.points) != len(otherContour.points):
+            return False
+
+        # Handle the two cases of samePointType
+        if samePointType:
+            selfPoints = [(point.position, point.type) for point in self.points]
+            otherPoints = [(point.position, point.type) for point in otherContour.points]
+        else:
+            selfPoints = [point.position for point in self.points]
+            otherPoints = [point.position for point in otherContour.points]
+
+        if sameStartPoint and samePosition:
+            return selfPoints == otherPoints
+        elif not sameStartPoint and samePosition:
```

Comment:
Perhaps this should go, as one could have a two contours with the same points and number of points but different direction or shape (depending on how the contours were connected). The next bit should cover this, I think.